homeassistant/components/sensor/forecast.py

```diff
@@ -124,78 +116,78 @@ def state(self):
-            elif self.type == 'precip_probability':
-                self._state = round(data.precipProbability * 100, 1)
-            elif self.type == 'dew_point':
-                self._state = round(data.dewPoint, 1)
-            elif self.type == 'temperature':
-                self._state = round(data.temperature, 1)
-            elif self.type == 'apparent_temperature':
-                self._state = round(data.apparentTemperature, 1)
-            elif self.type == 'wind_speed':
-                self._state = data.windSpeed
-            elif self.type == 'wind_bearing':
-                self._state = data.windBearing
-            elif self.type == 'cloud_cover':
-                self._state = round(data.cloudCover * 100, 1)
-            elif self.type == 'humidity':
-                self._state = round(data.humidity * 100, 1)
-            elif self.type == 'pressure':
-                self._state = round(data.pressure, 1)
-            elif self.type == 'visibility':
-                self._state = data.visibility
-            elif self.type == 'ozone':
-                self._state = round(data.ozone, 1)
-
+            else:
+                if self.type == 'summary':
+                    self._state = data.summary
+                elif self.type == 'icon':
+                    self._state = data.icon
+                elif self.type == 'nearest_storm_distance':
+                    self._state = data.nearestStormDistance
+                elif self.type == 'nearest_storm_bearing':
+                    self._state = data.nearestStormBearing
+                elif self.type == 'precip_intensity':
+                    self._state = data.precipIntensity
+                elif self.type == 'precip_type':
+                    self._state = data.precipType
+                elif self.type == 'precip_probability':
+                    self._state = round(data.precipProbability * 100, 1)
+                elif self.type == 'dew_point':
+                    self._state = round(data.dewPoint, 1)
+                elif self.type == 'temperature':
+                    self._state = round(data.temperature, 1)
+                elif self.type == 'apparent_temperature':
+                    self._state = round(data.apparentTemperature, 1)
+                elif self.type == 'wind_speed':
+                    self._state = data.windSpeed
+                elif self.type == 'wind_bearing':
+                    self._state = data.windBearing
+                elif self.type == 'cloud_cover':
+                    self._state = round(data.cloudCover * 100, 1)
+                elif self.type == 'humidity':
+                    self._state = round(data.humidity * 100, 1)
+                elif self.type == 'pressure':
+                    self._state = round(data.pressure, 1)
+                elif self.type == 'visibility':
+                    self._state = data.visibility
+                elif self.type == 'ozone':
+                    self._state = round(data.ozone, 1)
```

Comment:
I don't think that there are strong opinions on how things should done beside that it follows common Python standards and is readable. Often the code is easier to understand if there are some extra lines and no fancy shortcuts. But yes, saving 20 lines by doing something we already to do at a lot of other places (using dicts to do lookups) is definitely OK from my point of view.

This code segment here is more or less a copy of the OWM sensor. And that one uses the same style as the systemmonitor sensor.